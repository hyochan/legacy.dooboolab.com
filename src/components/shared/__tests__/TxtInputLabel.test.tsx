import React from 'react';
import { shallow } from 'enzyme';

import TxtInputLabel from '../TxtInputLabel';

describe('TxtInputLabel component test', () => {
  const component = shallow(
    <TxtInputLabel
      style={{ marginTop: 18 }}
      label='EMAIL'
      value='mock-email'
      placeholder='email'
      onTxtChanged={ () => { return; }}
    />,
  );

  it('component matches with its snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('first children prop has set correctly', () => {
    expect(component.props().children[0].props.children).toEqual('EMAIL');
  });

  describe('component is triggered with mock argument', () => {
    it('second children prop changes value', () => {
      const input = component.find('input');
      input.simulate('change', {
        target: {
          value: 'mock-email',
        },
      });
      expect(component.props().children[1].props.value).toEqual('mock-email');
    });
  });
});
