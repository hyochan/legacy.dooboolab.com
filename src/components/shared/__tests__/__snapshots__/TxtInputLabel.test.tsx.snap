// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxtInputLabel component test component matches with its snapshot 1`] = `ShallowWrapper {}`;
